pkg/controller/compliancescan/compliancescan_controller.go

```diff
@@ -384,6 +405,136 @@ func (r *ReconcileComplianceScan) phaseLaunchingHandler(h scanTypeHandler, logge
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      getKubeletCMNameForScan(instance, node),
+			Namespace: instance.Namespace,
+			CreationTimestamp: metav1.Time{
+				Time: time.Now(),
+			},
+			Labels: map[string]string{
+				compv1alpha1.ComplianceScanLabel: instance.Name,
+			},
+		},
+		Data: map[string]string{
+			KubeletConfigMapName: kubeletConfig,
+		},
+	}
+	err = r.Client.Create(context.TODO(), kubeletConfigCM)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *ReconcileComplianceScan) getNodeForScan(instance *compv1alpha1.ComplianceScan) (corev1.NodeList, error) {
+	nodes := corev1.NodeList{}
+	nodeScanSelector := map[string]string{"kubernetes.io/os": "linux"}
+	listOpts := client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(labels.Merge(instance.Spec.NodeSelector, nodeScanSelector)),
+	}
+
+	if err := r.Client.List(context.TODO(), &nodes, &listOpts); err != nil {
+		return nodes, err
+	}
+	return nodes, nil
+}
+
+// checkConfigMapTimestamp checks if the configmap is older than the scan start time
+// if it is, then this will return true, else false
+func (r *ReconcileComplianceScan) checkConfigMapTimestamp(instance *compv1alpha1.ComplianceScan, node *corev1.Node) (bool, error) {
+	kubeletConfigCM := &corev1.ConfigMap{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: getKubeletCMNameForScan(instance, node), Namespace: instance.Namespace}, kubeletConfigCM)
+	if err != nil {
+		return false, err
+	}
+	// check if the configmap is older the scan start time
+	if instance.Status.StartTimestamp == nil {
+		return false, fmt.Errorf("scan start timestamp is nil")
+	}
+	if kubeletConfigCM.CreationTimestamp.Before(instance.Status.StartTimestamp) {
+		return true, nil
+	}
+	return false, nil
+}
+
+func (r *ReconcileComplianceScan) handleRuntimeKubeletConfig(instance *compv1alpha1.ComplianceScan, logger logr.Logger) error {
+	// only handle node scans
+	if instance.Spec.ScanType != compv1alpha1.ScanTypeNode {
+		return nil
+	}
+	nodes, err := r.getNodeForScan(instance)
+	if err != nil {
+		logger.Error(err, "Cannot get the nodes for the scan")
+		return err
+	}
+	for _, node := range nodes.Items {
+		// check if the configmap already exists for the node
+		kubeletConfigCM := &corev1.ConfigMap{}
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Name: getKubeletCMNameForScan(instance, &node), Namespace: instance.Namespace}, kubeletConfigCM)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				logger.Error(err, "Error getting the configmap for the runtime kubeletconfig", "node", node.Name)
+				return err
+			}
+			// create a configmap for the node
+			err = r.createKubeletConfigCM(instance, &node, "")
+			logger.Info("Created a new configmap for the node", "configmap", kubeletConfigCM.Name, "node", node.Name)
+			if err != nil {
+				logger.Error(err, "Error creating the configmap for the runtime kubeletconfig", "node", node.Name)
+				return err
+			}
+			continue
```

Comment:
So even if we can't get the config, the scan still proceeds?

Reply:
If the error is not not found, we will return error and not continue, otherwise we create the new configmap